File: BOJ(Java 공부용)/1747.java
import java.io.*;
import static java.lang.Math.sqrt;
public class Main {
    static BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));
    static int ans = 0;
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int n = Integer.parseInt(br.readLine().split(" ")[0]);
        int[] notprime = new int[1004000];
        notprime[1] = 1;
        for(int i=2;i<=sqrt(1004000);i++) {
            for(int j=i*i;j<1004000;j+=i) {
                notprime[j] = 1;
            }
        }
        boolean ok = false;
        for(int i=n;;i++) {
            if(notprime[i] != 0) continue;
            String s = String.valueOf(i);
            int temp = 0;
            for(int j=0,k=s.length()-1;j<=k;j++,k--) {
                if(s.charAt(j) != s.charAt(k)) {
                    temp++;
                    break;
                }
             }
            if(temp == 0) {
                ok = true;
            }
            if(ok) {
                System.out.println(i);
                return;
            }
        }
    }
}

File: BOJ(Python 공부용)/1747.py
import sys
from math import sqrt

input = sys.stdin
sys.setrecursionlimit(100)

notprime = [False] * 1004000
notprime[1] = True
for i in range(2, int(sqrt(1004000))):
    for j in range(i * i, 1004000, i):
        notprime[j] = True
n = int(input.readline())
ok = False
for i in range(n, 1004000, 1):
    if notprime[i]: continue
    s = str(i)
    temp = False
    for j, k in zip(range(0, len(s), 1), range(len(s) - 1, 0, -1)):
        if j > k: break
        if s[j] != s[k]:
            temp = True
            break

    if not temp:
        print(i)
        break
